src/filters/fs.rs

```diff
@@ -369,23 +369,21 @@ fn bytes_range(range: Option<Range>, max_len: u64) -> Result<(u64, u64), BadRang
     let ret = range
         .iter()
         .map(|(start, end)| {
-            let start = match start {
-                Bound::Unbounded => 0,
-                Bound::Included(s) => s,
-                Bound::Excluded(s) => s + 1,
-            };
-
-            let end = match end {
-                Bound::Unbounded => max_len,
-                Bound::Included(s) => {
+            let (start, end) = match (start, end) {
+                (Bound::Unbounded, Bound::Unbounded) => (0, max_len),
+                (Bound::Included(b), Bound::Included(e)) => {
                     // For the special case where s == the file size
-                    if s == max_len {
-                        s
-                    } else {
-                        s + 1
+                    let e = if e == max_len { e } else { e + 1 };
+                    (b, e)
+                }
+                (Bound::Included(b), Bound::Unbounded) => (b, max_len),
+                (Bound::Unbounded, Bound::Included(e)) => {
+                    if e > max_len {
+                        return Err(BadRange);
                     }
+                    (max_len - e, max_len)
                 }
-                Bound::Excluded(s) => s,
+                _ => unreachable!(),
```

Comment:
`Range::iter()`  [only returns `Bounds::included` or `Bounds::Unbounded`](https://github.com/hyperium/headers/blob/master/src/common/range.rs#L71-L84)

Reply:
Rust says that `unreachable!()` always panic so should we avoid it with something like `return Err(BadRange);`? what do you think?

Reply:
I got it! I think my previous comment is irrelevant since all cases are covered and `unreachable!()` just clarify the last case for compiler.